### ISSUE TITLE
Regularizion loss in spatial_aware not actually used

### ISSUE THREAD

**Issue by Miaowshroom** [YujiaoShi/cross_view_localization_SAFA#7]
Thanks for sharing the code of SAFA.

I am trying to re-implemente it in Pytorch. Getting confused about the regularizer of `w1, b1, w2, b2 `of `spatial_aware`.

The regularizer is defined in the variable:

> bias1 = tf.get_variable(name='biases1', shape=[1, int(height * width/2), dimension], trainable=trainable,      
               initializer=tf.constant_initializer(0.1), regularizer=tf.contrib.layers.l1_regularizer(0.01))

However, the reg loss is not added for the optimizer:

> train_step = tf.train.AdamOptimizer(learning_rate, 0.9, 0.999).minimize(loss, global_step=global_step)
> loss = compute_loss(sat_global, grd_global)

according to https://stackoverflow.com/questions/37107223/how-to-add-regularizations-in-tensorflow
The reg loss is collected but not automatically added in the overall loss 
Shall add `loss = loss + tf.losses.get_regularization_loss()`

So is this mean the SAFA did not actually learn from the regularisation loss?

**Comment by shiyujiao** [comment]
Yes, I think so.

**Comment by YujiaoShi** [comment]
> Thanks for sharing the code of SAFA.
> 
> I am trying to re-implemente it in Pytorch. Getting confused about the regularizer of `w1, b1, w2, b2 `of `spatial_aware`.
> 
> The regularizer is defined in the variable:
> 
> > bias1 = tf.get_variable(name='biases1', shape=[1, int(height * width/2), dimension], trainable=trainable,
> > initializer=tf.constant_initializer(0.1), regularizer=tf.contrib.layers.l1_regularizer(0.01))
> 
> However, the reg loss is not added for the optimizer:
> 
> > train_step = tf.train.AdamOptimizer(learning_rate, 0.9, 0.999).minimize(loss, global_step=global_step)
> > loss = compute_loss(sat_global, grd_global)
> 
> according to https://stackoverflow.com/questions/37107223/how-to-add-regularizations-in-tensorflow The reg loss is collected but not automatically added in the overall loss Shall add `loss = loss + tf.losses.get_regularization_loss()`
> 
> So is this mean the SAFA did not actually learn from the regularisation loss?

Yes, I think so.
